token/program-2022/src/extension/mod.rs

```diff
@@ -159,7 +159,12 @@ fn get_extension_indices<V: Extension>(
     Err(ProgramError::InvalidAccountData)
 }
 
-fn get_extension_types(tlv_data: &[u8]) -> Result<Vec<ExtensionType>, ProgramError> {
+/// Fetches the extension types written to the buffer, along with the total
+/// number of bytes used by the TLV entries, including the type and length of
+/// each extension.
+fn get_extension_types_and_len(
+    tlv_data: &[u8],
+) -> Result<(Vec<ExtensionType>, usize), ProgramError> {
```

Comment:
I know the return-type tuple only has 2 elements, but I think it might be nicer as a struct with named fields.

Reply:
No problem, I think it neatly resolves the problems with the unclear function comment

Reply:
Much clearer, thank you for taking the trouble!